imap_processing/utils.py

```diff
@@ -122,8 +127,21 @@ def create_dataset(packets: list[Packet], met_name="shcoarse") -> xr.Dataset:
     description_dict = {}
 
     for data_packet in packets:
+        current_data = (
+            (data_packet.header | data_packet.data)
+            if include_header
+            else data_packet.data
+        )
+
+        # Drop keys using skip_keys
+        if skip_keys is not None:
+            current_data = {
```

Comment:
This is very minor considering a relatively small size of the `current_data` dict, but this would modify the dictionary in place rather than replicating a new one.
```suggestion
            for key in skip_keys:
                current_data.pop(key, None)
```

Reply:
Thank you!